File: project/src/main/java/com/trojanscheduler/project/model/Section.java
package com.trojanscheduler.project.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Entity(name="sections")
@Data
public class Section {
    @Id
    private int id;

    @ManyToOne
    @JoinColumn(name="course_id")
    @JsonIgnore
    private Course course;

    @ManyToMany(mappedBy="sectionList")
    @JsonIgnore
    private Set<TrojanUser> users = new HashSet<>();

    private String section;
    private String session;
    private String type;
    private int startTime;
    private int endTime;
    private int days;
    private String registered;
    private String instructor;
    private String location;

    public int hashCode() {
        return Objects.hash(id, section);
    }





}

File: get_course_data/model/section.py
import dataclasses


@dataclasses.dataclass
class Section:
    section: str = ""
    session: str = ""
    type: str = ""
    start_time: int = 0
    end_time: int = 0
    days: int = 0
    registered: str = ""
    instructor: str = ""
    location: str = ""

    # instructor_link: str = ""
    # syllabus: str = ""
    # info: str = ""

    def to_dict(self):
        return {
            "section": self.section,
            "session": self.session,
            "type": self.type,
            "start_time": self.start_time,
            "end_time": self.end_time,
            "days": self.days,
            "registered": self.registered,
            "instructor": self.instructor,
            "location": self.location
        }

    def __str__(self):
        # return str(self.to_dict())
        return f"{self.section} {self.session} {self.type} {self.start_time} {self.end_time} {self.days} {self.registered} {self.instructor} {self.location}"


